astropy/coordinates/builtin_frames/fk4.py

```diff
@@ -206,4 +206,4 @@ def fk4_no_e_to_fk4(fk4noecoord, fk4frame):
     if isinstance(fk4noecoord.data, UnitSphericalRepresentation):
```

Comment:
Here, it seems more logical to just do `rep = rep.represent_as(fk4noecoord.representation_type)` for all cases. And perhaps one should do the same elsewhere too?
Alternatively, I think the `if` clause can just be deleted as you now give `representation_type` below.

Reply:
There are several instances of `if isinstance\(.*, UnitSpherical` where different logic is applied if there is no distance information, so I only replaced this one that you pointed out as well as the inverse one.

Reply:
...I misunderstood the comment, now I see that I can actually remove the `if` clause. Doing so.